Adafruit_BNO055.cpp

```diff
@@ -549,6 +586,458 @@ bool Adafruit_BNO055::isFullyCalibrated(void)
+      } else {
+        status = write8(BNO055_INTR_MSK_ADDR, (read8(BNO055_INTR_MSK_ADDR)) | (0 << 7)); // enable pin change
+      }
+      write8(BNO055_INTR_EN_ADDR, (read8(BNO055_INTR_EN_ADDR)) | (1 << 7)); // enable interrupt
+      write8(BNO055_INTR_ACCEL_NM_SETT, (read8(BNO055_INTR_ACCEL_NM_SETT)) | (0 << 0)); // decides SM/NM
+    break;
+    case ACC_AM:
+      if(triggerPin){
+        status = write8(BNO055_INTR_MSK_ADDR, (read8(BNO055_INTR_MSK_ADDR)) | (1 << 6)); // enable pin change
+      } else {
+        status = write8(BNO055_INTR_MSK_ADDR, (read8(BNO055_INTR_MSK_ADDR)) | (0 << 6)); // enable pin change
+      }
+      write8(BNO055_INTR_EN_ADDR, (read8(BNO055_INTR_EN_ADDR)) | (1 << 6)); // enable interrupt
+    break;
+    case ACC_SM:
+      if(triggerPin){
+        status = write8(BNO055_INTR_MSK_ADDR, (read8(BNO055_INTR_MSK_ADDR)) | (1 << 7)); // enable pin change
+      } else {
+        status = write8(BNO055_INTR_MSK_ADDR, (read8(BNO055_INTR_MSK_ADDR)) | (0 << 7)); // enable pin change
+      }
+      write8(BNO055_INTR_EN_ADDR, (read8(BNO055_INTR_EN_ADDR)) | (1 << 7)); // enable interrupt
+      write8(BNO055_INTR_ACCEL_NM_SETT, (read8(BNO055_INTR_ACCEL_NM_SETT)) | (1 << 0)); // decides SM/NM
+    break;
+    case ACC_HIGH_G:
+      if(triggerPin){
+        status = write8(BNO055_INTR_MSK_ADDR, (read8(BNO055_INTR_MSK_ADDR)) | (1 << 5)); // enable pin change
+      } else {
+        status = write8(BNO055_INTR_MSK_ADDR, (read8(BNO055_INTR_MSK_ADDR)) | (0 << 5)); // enable pin change
+      }
+      write8(BNO055_INTR_EN_ADDR, (read8(BNO055_INTR_EN_ADDR)) | (1 << 5)); // enable interrupt
+    break;
+    case GYR_HIGH_RATE:
+      if(triggerPin){
+        status = write8(BNO055_INTR_MSK_ADDR, (read8(BNO055_INTR_MSK_ADDR)) | (1 << 3)); // enable pin change
+      } else {
+        status = write8(BNO055_INTR_MSK_ADDR, (read8(BNO055_INTR_MSK_ADDR)) | (0 << 3)); // enable pin change
+      }
+      write8(BNO055_INTR_EN_ADDR, (read8(BNO055_INTR_EN_ADDR)) | (1 << 3)); // enable interrupt
+    break;
+    case GYRO_AM:
+      if(triggerPin){
+        status = write8(BNO055_INTR_MSK_ADDR, (read8(BNO055_INTR_MSK_ADDR)) | (1 << 2)); // enable pin change
+      } else {
+        status = write8(BNO055_INTR_MSK_ADDR, (read8(BNO055_INTR_MSK_ADDR)) | (0 << 2)); // enable pin change
+      }
+      write8(BNO055_INTR_EN_ADDR, (read8(BNO055_INTR_EN_ADDR)) | (1 << 2)); // enable interrupt
+    break;
+  }
+  // restore page ID
+  write8(BNO055_PAGE_ID_ADDR, savePageID);
+
+  // Set the requested operating mode (see section 3.3)
+  setMode(modeback);
+  delay(20);
+
+  if (status > 0){
+    return false;
+  } else {
+    return true;
+  }
+
+}
+
+/**************************************************************************/
+/*!
+@brief  Sets interrupt axes
+*/
+/**************************************************************************/
+
+bool Adafruit_BNO055::enableInterruptAxes( adafruit_bno055_intr_en_t int_en_code, String axes )
```

Comment:
Using an enum or a bit field (a `uint8_t` where `x = 0b00000001`, `y = 0b00000010`, `z  0b00000100` or `1<<2`, etc.) would probably be more natural for embedded systems, and less work for the MCU to manipulate???